astropy/time/tests/test_quantity_interaction.py

```diff
@@ -216,3 +216,19 @@ def test_delta_tdb_tt(self):
         assert t.tdb.iso == '2010-01-01 00:00:40.000000'
         t.delta_tdb_tt = TimeDelta(50./24./3600., format='jd')
         assert t.tdb.iso == '2010-01-01 00:00:50.000000'
+
+
+@pytest.mark.parametrize('q', (5e8*u.s, 5e17*u.ns))
+def test_quantity_conversion_rounding(q):
```

Comment:
It doesn't look like there is any coverage of the case where `val2` is not None.

Reply:
Oops, corrected.